examples/advanced/spot_instances/aws/main.tf

```diff
@@ -0,0 +1,76 @@
+terraform {
+  required_version = ">= 0.14.2"
+}
+
+module "aws" {
+  source         = "../../../../aws"
+  config_git_url = "https://github.com/ComputeCanada/puppet-magic_castle.git"
+  config_version = "main"
+
+  cluster_name = "phoenix"
+  domain       = "calculquebec.cloud"
+  image        = "ami-033e6106180a626d0" # CentOS 7 -  ca-central-1
+
+  instances = {
+    mgmt  = { type = "t3.large",  count = 1, tags = ["mgmt", "puppet", "nfs"] },
+    login = { type = "t3.medium", count = 1, tags = ["login", "public", "proxy"] },
+    node  = { 
+        type = "t3.medium",
+        count = 1,
+        tags = ["node", "spot"],
+    }
+  }
+
+  volumes = {
+    nfs = {
+      home     = { size = 10, type = "gp2" }
+      project  = { size = 50, type = "gp2" }
+      scratch  = { size = 50, type = "gp2" }
+    }
+  }
+
+  public_keys = [file("~/.ssh/id_rsa.pub")]
+
+  nb_users     = 10
+  # Shared password, randomly chosen if blank
+  guest_passwd = ""
+
+  # AWS specifics
```

Comment:
Would the spot specific variables be added here?

Reply:
spot variables are per instance, so they are defined in the instance attributes.

Reply:
Ah sure, ok